coalib/settings/FunctionMetadata.py

```diff
@@ -160,3 +160,53 @@ def from_function(cls, func, omit=frozenset()):
                    non_optional_params=non_optional_params,
                    optional_params=optional_params,
                    omit=omit)
+
+    @classmethod
+    def merge(cls, *metadatas):
+        """
+        Merges signatures of ``FunctionMetadata`` objects.
+
+        Parameter descriptions (either optional or non-optional) are merged
+        from left to right, meaning the right hand metadata overrides the left
+        hand one.
+
+        :param metadatas:
+            The sequence of metadatas to merge.
+        :raises ValueError:
+            Raised when less than 2 metadatas are provided.
+        :return:
+            A ``FunctionMetadata`` object containing the merged signature of
+            all given ones.
+        """
+        # Collect the metadatas, as we operate on them more often and we want
+        # to support arbitrary sequences.
+        metadatas = tuple(metadatas)
+        if len(metadatas) < 2:
+            raise ValueError("Less than 2 metadata objects provided.")
+
+        merged_name = ("<Merged signature of " +
+                       ", ".join(repr(metadata.name)
+                                 for metadata in metadatas) +
+                       ">")
+
+        merged_desc = "\n".join("{}:\n{}".format(metadata.name, metadata.desc)
+                                for metadata in metadatas)
```

Comment:
The descriptions are strings right? So I can't easily extract the param-contents (or other attributes).
But this was in fact a little "prototype", I wasn't completely sure how a clean "merge" function should look like^^ maybe we can discuss on gitter^^